src/system/SystemPacketBuffer.cpp

```diff
@@ -207,25 +208,36 @@ void PacketBuffer::SetStart(uint8_t * aNewStart)
         aNewStart = kEnd;
 
     ptrdiff_t lDelta = aNewStart - static_cast<uint8_t *>(this->payload);
-    if (lDelta > this->len)
-        lDelta = this->len;
+    if (lDelta > static_cast<int32_t>(this->len))
+        lDelta = static_cast<int32_t>(this->len);
```

Comment:
What I would expect is that we would check for negative lDelta, perhaps, but then cast lDelta to size_t to do the compare, instead of casting our length.